pkg/containerutil/containerutil.go

```diff
@@ -463,10 +466,96 @@ func Unpause(ctx context.Context, client *containerd.Client, id string) error {
 	}
 }
 
-// Returns the path to the Nerdctl-managed state directory for the container with the given ID.
-func ContainerStateDirPath(globalOptions types.GlobalCommandOptions, dataStore, id string) (string, error) {
-	if err := nsutil.ValidateNamespaceName(globalOptions.Namespace); err != nil {
-		return "", fmt.Errorf("invalid namespace name %q for determining state dir of container %q: %s", globalOptions.Namespace, id, err)
+// ContainerStateDirPath returns the path to the Nerdctl-managed state directory for the container with the given ID.
+func ContainerStateDirPath(ns, dataStore, id string) (string, error) {
+	if err := nsutil.ValidateNamespaceName(ns); err != nil {
+		return "", fmt.Errorf("invalid namespace name %q for determining state dir of container %q: %s", ns, id, err)
+	}
+	return filepath.Join(dataStore, "containers", ns, id), nil
+}
+
+// WriteCIDFile writes the container ID to a file.
+func WriteCIDFile(path, id string) error {
+	if _, err := os.Stat(path); err == nil {
+		return fmt.Errorf("container ID file found, make sure the other container isn't running or delete %s", path)
+	} else if errors.Is(err, os.ErrNotExist) {
+		f, err := os.Create(path)
+		if err != nil {
+			return fmt.Errorf("failed to create the container ID file: %s for %s, err: %s", path, id, err)
+		}
+		defer f.Close()
+
+		if _, err := f.WriteString(id); err != nil {
+			return err
+		}
+		return nil
+	} else {
+		return err
+	}
+}
+
+// CheckRestartCapabilities checks the capabilities of the restart plugin.
+func checkRestartCapabilities(ctx context.Context, client *containerd.Client, restartFlag string) (bool, error) {
```

Comment:
Is this func used in places other than `restart_options.go`? If not, it can be moved there to collocate related logic.